pkg/v1/tkg/web/src/app/views/landing/wizard/shared/components/steps/metadata-step/metadata-step.fieldmapping.ts

```diff
@@ -1,19 +1,49 @@
-import { StepMapping } from '../../../field-mapping/FieldMapping';
-import { SimpleValidator } from '../../../constants/validation.constants';
+import {ControlType, StepMapping} from '../../../field-mapping/FieldMapping';
+import {SimpleValidator} from '../../../constants/validation.constants';
 
 export enum MetadataField {
     CLUSTER_LABELS = 'clusterLabels',
     CLUSTER_DESCRIPTION = 'clusterDescription',
-    CLUSTER_LOCATION = 'clusterLocation'
+    CLUSTER_LOCATION = 'clusterLocation',
 }
 
 export const MetadataStepMapping: StepMapping = {
     fieldMappings: [
-        { name: MetadataField.CLUSTER_LOCATION, validators: [SimpleValidator.IS_VALID_LABEL_OR_ANNOTATION], label: 'LOCATION (OPTIONAL)' },
-        { name: MetadataField.CLUSTER_DESCRIPTION, validators: [SimpleValidator.IS_VALID_LABEL_OR_ANNOTATION], label: 'DESCRIPTION (OPTIONAL)' },
-        { name: MetadataField.CLUSTER_LABELS, hasNoDomControl: true, isMap: true, label: 'LABELS (OPTIONAL)' },
-    ]
-}
+        {
+            name: MetadataField.CLUSTER_LOCATION,
+            validators: [SimpleValidator.IS_VALID_LABEL_OR_ANNOTATION],
+            label: 'LOCATION (OPTIONAL)',
+        },
+        {
+            name: MetadataField.CLUSTER_DESCRIPTION,
+            validators: [SimpleValidator.IS_VALID_LABEL_OR_ANNOTATION],
+            label: 'DESCRIPTION (OPTIONAL)',
+        },
+        {
+            name: MetadataField.CLUSTER_LABELS,
+            label: 'LABELS (OPTIONAL)',
+            controlType: ControlType.FormArray,
+            children: [
+                {
+                    name: 'key',
+                    defaultValue: '',
+                    controlType: ControlType.FormControl,
+                    validators: [
+                        SimpleValidator.IS_VALID_LABEL_OR_ANNOTATION,
+                        SimpleValidator.RX_UNIQUE,
```

Comment:
If we can avoid introducing another library for validation, I think that is best. How much effort do you think would be involved for writing a couple of custom validator methods in validation.service.ts? I think if you're able to leverage lodash's _.uniq or _.uniqBy method, it might not be too bad. Do you mind giving it a quick look if you haven't already? We can chat over it more in the UI team sync if you'd like

Reply:
as discussed we are going to keep this